components/provisioning/org.wso2.carbon.identity.provisioning.connector.google/src/main/java/org/wso2/carbon/identity/provisioning/connector/google/GoogleProvisioningConnector.java

```diff
@@ -76,15 +77,16 @@ public void init(Property[] provisioningProperties) throws IdentityProvisioningE
             for (Property property : provisioningProperties) {
 
                 if (GoogleConnectorConstants.PRIVATE_KEY.equals(property.getName())) {
+                    FileOutputStream fos = null;
                     try {
                         byte[] decodedBytes = Base64Utils.decode(property.getValue());
                         googlePrvKey = new File("googlePrvKey");
-                        FileOutputStream fos = new FileOutputStream(googlePrvKey);
+                        fos = new FileOutputStream(googlePrvKey);
                         fos.write(decodedBytes);
-                        fos.flush();
```

Comment:
flush() is removed ? Better have flush() because you can find that some JDKs swallow the exception if flush() is done inside the close(). So better keep it.